dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -398,9 +399,21 @@ public Workflow setupWorkflowVersions(String repositoryId, Workflow workflow, Op
         // For each branch (reference) found, create a workflow version and find the associated descriptor files
         for (Triple<String, Date, String> ref : references) {
             if (ref != null) {
-                WorkflowVersion version = setupWorkflowVersionsHelper(workflow, ref, existingWorkflow, existingDefaults,
-                    repository, null, versionName);
-                if (version != null) {
+                if (toRefreshVersion(ref.getLeft(), ref.getRight(), existingDefaults, hardRefresh)) {
+                    WorkflowVersion version = setupWorkflowVersionsHelper(workflow, ref, existingWorkflow, existingDefaults,
+                            repository, null, versionName);
+                    if (version != null) {
+                        workflow.addWorkflowVersion(version);
+                    }
+                } else {
+                    // Version didn't change, but we don't want to delete
+                    // Add a stub version with commit ID set to an ignore value
```

Comment:
Probably missed the plot, when/why does this SKIP_COMMIT_ID come into play?
(might be good code comment)

Reply:
So if we didn't include this stubbed version with the commit ID as SKIP_COMMIT_ID and simply only looked at the versions that are being refreshed, they are deleted later on since later code assumes that any version that is no longer present in the new set of refreshed versions is to be deleted.

Reply:
Ok, so if I understand correctly, in other words, these stubbed versions correspond to real previously existing versions that we don't want to delete. So we create this stub to hold their place.

Reply:
Yes that is exactly it. I can add a comment to the code.